Tests/Functional/Bundle/TestBundle/Controller/ParamFetcherController.php

```diff
@@ -54,4 +56,54 @@ public function testAction(Request $request, ParamFetcherInterface $fetcher)
             'after' => $paramsAfter,
         ));
     }
+
+    /**
+     * @FileParam(name="single_file", strict=false, default="noFile")
+     */
+    public function singleFileAction(ParamFetcherInterface $fetcher)
+    {
+        /** @var UploadedFile $file */
+        $file = $fetcher->get('single_file');
+
+        return new JsonResponse(array(
+            'single_file' => $this->printFile($file),
+        ));
+    }
+
+    /**
+     * @FileParam(name="array_files", map=true)
+     */
+    public function fileCollectionAction(ParamFetcherInterface $fetcher)
+    {
+        $files = $fetcher->get('array_files');
+
+        return new JsonResponse(array(
+            'array_files' => [
+                $this->printFile($files[0]),
+                $this->printFile($files[1]),
+            ],
+        ));
+    }
+
+    /**
+     * @FileParam(name="array_images", image=true, strict=false, map=true, default="NotAnImage")
+     */
+    public function imageCollectionAction(ParamFetcherInterface $fetcher)
+    {
+        $files = $fetcher->get('array_images');
+
+        return new JsonResponse(array(
+            'array_files' => (is_string($files)) // Default message on validation error
```

Comment:
Yes, but it could lead to confusion when someone is looking at the tests to understand how the feature works.